lib/typinggame_server/repositories/player_room_repository.rb

```diff
@@ -1,7 +1,19 @@
 class PlayerRoomRepository < Hanami::Repository
+  associations { has_many :players }
+
   def find_player_room_records(player_id: nil, room_id:)
     return player_rooms.where(room_id: room_id).to_a if player_id.nil?
 
     player_rooms.where(player_id: player_id).where(room_id: room_id).to_a.first
   end
+
+  def find_players_names_positions(room_id:)
+    player_rooms.read(
+      "SELECT players.name, player_rooms.position, player_rooms.player_id FROM player_rooms JOIN players ON players.id = player_rooms.player_id WHERE room_id = #{
```

Comment:
I assume this is how Hanami recommends writing SQL in order to avoid SQL injection attacks?

Reply:
[Yeah I imagine so, as seen here in their example.](https://guides.hanamirb.org/repositories/sql-queries/#raw-sql)

Unfortunately I spent like 5-6 hours trying to figure out how to do this using the [ROM that Hanami provides](https://rom-rb.org/) and I couldn't figure it out. I wasn't the only person with this confusion either:

https://stackoverflow.com/questions/43080678/join-query-in-hanami-model
https://stackoverflow.com/questions/50011179/how-to-fetch-data-from-multiple-tables-with-one-sql-query-in-hanami

Neither of these answers got the job done, but I want to re-visit it at some point because the ideas conveyed make sense.